src/cpufreq.c

```diff
@@ -18,48 +18,238 @@
  *
  * Authors:
  *   Peter Holik <peter at holik.at>
+ *   rinigus <http://github.com/rinigus>
+ *
+ * CPU frequency is recorded either as a current value or as changes
+ * in used CPU frequency distribution. For current value,
+ * "/sys/devices/system/cpu/cpuX/cpufreq/scaling_cur_freq" are
+ * used. For reporting distribution changes, amount of used time on
+ * each supported CPU frequency is reported using DERIVE
+ * type. Distributions are collected from
+ * "/sys/devices/system/cpu/cpuX/cpufreq/stats/time_in_state", format
+ * description at https://www.kernel.org/doc/Documentation/cpu-freq/cpufreq-stats.txt.
+ *
+ * Units of reported values:
+ *
+ *  Frequencies in Hz when reported as a current value
+ *  Frequencies in kHz when used as a type instance for distributions
+ *  Distribution utilization for each CPU frequency in milliseconds that this frequency was
+ *   used per second of wall time
+ *
  **/
 
 #include "collectd.h"
-
```

Comment:
Please leave this line here. It's there so that clang-format keeps this header on top, see a5377cf